index/secondary_index_manager.cc

```diff
@@ -156,7 +155,7 @@ sstring index_table_name(const sstring& index_name) {
 }
 
 sstring index_name_from_table_name(const sstring& table_name) {
-    if (table_name.size() < 7 || !boost::algorithm::ends_with(table_name, "_index")) {
+    if (table_name.size() < 7 || table_name.ends_with("_index")) {
```

Comment:
The size is [checked](https://en.cppreference.com/w/cpp/string/basic_string_view/ends_with) internally, no need to check here:
```suggestion
    if (table_name.ends_with("_index")) {
```

Reply:
indeed.

Reply:
if we change as you suggest, that'd change existing behaviour. please note, we throw if `table_name.size() < 7` instead of  `table_name.size() >= 6`.

Reply:
Presumably the new code should be `... || !table_name.ends_with("_index")) {` (the negation is lost)

Reply:
thanks. i missed it.